arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/SstoreSection.java

```diff
@@ -46,21 +49,21 @@ public class SstoreSection extends TraceSection implements PostRollbackDefer {
   final EWord valueCurrent;
   final EWord valueNext;
 
-  public SstoreSection(Hub hub, WorldView world) {
+  public SstoreSection(Hub hub, WorldView worldView) {
     super(
         hub,
         (short)
             (hub.opCode().numberOfStackRows() + (Exceptions.any(hub.pch().exceptions()) ? 5 : 4)));
 
-    this.world = world;
-    this.address = hub.messageFrame().getRecipientAddress();
-    this.deploymentNumber = hub.currentFrame().accountDeploymentNumber();
-    this.storageKey = Bytes32.leftPad(hub.messageFrame().getStackItem(0));
-    this.incomingWarmth = hub.messageFrame().getWarmedUpStorage().contains(address, storageKey);
-    this.valueOriginal =
-        EWord.of(world.get(address).getOriginalStorageValue(UInt256.fromBytes(storageKey)));
-    this.valueCurrent = EWord.of(world.get(address).getStorageValue(UInt256.fromBytes(storageKey)));
-    this.valueNext = EWord.of(hub.messageFrame().getStackItem(1));
+    world = worldView;
+    address = hub.messageFrame().getRecipientAddress();
+    deploymentNumber = hub.deploymentNumberOf(address);
+    storageKey = Bytes32.leftPad(hub.messageFrame().getStackItem(0));
+    incomingWarmth = hub.messageFrame().getWarmedUpStorage().contains(address, storageKey);
+    valueOriginal =
+        EWord.of(worldView.get(address).getOriginalStorageValue(UInt256.fromBytes(storageKey)));
+    valueCurrent = EWord.of(worldView.get(address).getStorageValue(UInt256.fromBytes(storageKey)));
+    valueNext = EWord.of(hub.messageFrame().getStackItem(1));
```

Comment:
Same remark about the `address of the underlying account` vs `address whose code is currently executing`.